src/auxinfo/auxinfo_commit.rs

```diff
@@ -123,22 +125,25 @@ impl CommitmentScheme {
         // Public parameters in this decommit must be consistent with each other
         scheme.clone().public_key.verify(context)?;
 
+        let lower_bound = BigNumber::one() << ELL;
+        assert!(scheme.public_key.params().scheme().modulus() >= &lower_bound);
+
         // Owner must be consistent across message, public keys, and decommit
         if scheme.public_key.participant() != scheme.pid {
             error!(
                 "Deserialized AuxInfoDecommit has different participant IDs in the sender ({}) and public_keys ({}) fields",
                 scheme.pid,
                 scheme.public_key.participant(),
             );
-            return Err(InternalError::ProtocolError);
+            return Err(InternalError::ProtocolError(Some(scheme.pid)));
         }
         if scheme.pid != message.from() {
             error!(
                 "Deserialized AuxInfoDecommit claiming to be from a different sender ({}) than the message was from ({})",
                 scheme.pid,
                 message.from()
             );
-            return Err(InternalError::ProtocolError);
+            return Err(InternalError::ProtocolError(Some(scheme.pid)));
```

Comment:
I think for these two cases, the "most correct" PID will be `message.from`; we require that the calling application check authentication for messages and maintain a mapping from PID to, like, whatever authentication details they need to validate stuff coming over the network. The only PID they'll have access to for that check will be the one in the message.

So, if a participant wants to cheat and lie about who owns the material they're sending, they can't lie about the `message.from` field, because that should get caught outside this library. They could lie about the scheme PID or the public key PID because those aren't tied to any other authentication mechanisms before this point.